Mathlib/Analysis/Calculus/LocalExtr/LineDeriv.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2024 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Analysis.Calculus.LocalExtr.Basic
+import Mathlib.Analysis.Calculus.LineDeriv.Basic
+
+/-!
+# Local extremum and line derivatives
+
+If `f` has a local extremum at a point, then the derivative at this point is zero.
+In this file we prove several versions of this fact for line derivatives.
+-/
+
+open Function Set Filter
+open scoped Classical Topology
+
+section Module
+
+variable {E : Type*} [AddCommGroup E] [Module ℝ E] {f : E → ℝ} {s : Set E} {a b : E} {f' : ℝ}
+
+theorem IsExtrFilter.hasLineDerivAt_eq_zero {l : Filter E} (h : IsExtrFilter f l a)
+    (hd : HasLineDerivAt ℝ f f' a b) (h' : Tendsto (fun t : ℝ ↦ a + t • b) (𝓝 0) l) : f' = 0 :=
+  IsLocalExtr.hasDerivAt_eq_zero (IsExtrFilter.comp_tendsto (by simpa using h) h') hd
+
+theorem IsExtrFilter.lineDeriv_eq_zero {l : Filter E} (h : IsExtrFilter f l a)
+    (h' : Tendsto (fun t : ℝ ↦ a + t • b) (𝓝 0) l) : lineDeriv ℝ f a b = 0 :=
+  if hd : LineDifferentiableAt ℝ f a b then h.hasLineDerivAt_eq_zero hd.hasLineDerivAt h'
+  else lineDeriv_zero_of_not_lineDifferentiableAt hd
```

Comment:
Is this our indentation convention for ite? I think it would be more legible as
```
  if hd : LineDifferentiableAt ℝ f a b then 
    h.hasLineDerivAt_eq_zero hd.hasLineDerivAt h'
  else 
    lineDeriv_zero_of_not_lineDifferentiableAt hd
```